static/js/tools/map/d3_map.tsx

```diff
@@ -75,39 +68,133 @@ const DEBOUNCE_INTERVAL_MS = 30;
 const DEFAULT_ZOOM_TRANSFORMATION = d3.zoomIdentity.scale(1).translate(0, 0);
 
 export function D3Map(props: D3MapProps): JSX.Element {
+  const { placeInfo, statVar, display } = useContext(Context);
+
   const [errorMessage, setErrorMessage] = useState("");
   const [zoomTransformation, setZoomTransformation] = useState(
     DEFAULT_ZOOM_TRANSFORMATION
   );
   const chartContainerRef = useRef<HTMLDivElement>();
 
-  function replot() {
```

Comment:
Here making draw() into a useCallback() effect, so it updates based on dependency changes, and ease the need for replot().